### PR TITLE
Api/articles

### PR DESCRIPTION
**Pull request by omar-baggio**

### CODE REVIEW

File: model/articles.model.js [hunk 1]
```diff
@@ -18,3 +18,18 @@ exports.selectArticleById = (article_id) => {
       return rows[0];
     });
 };
+
+exports.selectAllArticles = () => {
+  let sqlString = `
+  SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id) AS INT) AS comment_count
+  FROM articles
```
Comment (hevdevs) [inline model/articles.model.js:25]: Nice!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -104,3 +104,24 @@ describe("ERROR: GET /api/articles/:article_id", () => {
       });
   });
 });
+
+describe("GET /api/articles", () => {
+  test.only("status: 200, should responds with an article object, containing the properties author, title, article_id, body, topic, created_at, votes, article_img_url and comment_count", () => {
```
Comment (hevdevs) [inline __tests__/app.test.js]: remove .only in test

**Review by hevdevs — CHANGES_REQUESTED** [review]
This is looking great so far Omar! 🎉 
We still need one more test for this endpoint though. We another 200 test to assert that the response by default is sorted by date in descending order.
Have a look at jest-sorted to do this: https://npmjs.com/package/jest-sorted
Please put in another PR when done

File: __tests__/app.test.js [hunk 3]
```diff
@@ -104,3 +105,25 @@ describe("ERROR: GET /api/articles/:article_id", () => {
...
 });
+
+describe("GET /api/articles", () => {
+  test("status: 200, should responds with an article object, containing the properties author, title, article_id, body, topic, created_at, votes, article_img_url and comment_count", () => {
+    return request(app)
+      .get(`/api/articles`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toBeInstanceOf(Array);
+        expect(body).toBeSorted({ descending: true });
```
Comment (corkesw) [inline __tests__/app.test.js]: This doesn't look quite right, you should take a look at the docs for jest-sorted here...

https://www.npmjs.com/package/jest-sorted?activeTab=readme

The example when checking the sort order of an array of objects:


`expect([{ id: 1 }, { id: 2 }, { id: 3 }]).toBeSortedBy('id');`

**Comment by corkesw** [comment]
Hey Omar, nearly there! Just take a look at the jest-sorted docs again to see which matcher should be used and how.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -104,3 +105,28 @@ describe("ERROR: GET /api/articles/:article_id", () => {
...
+
+describe("GET /api/articles", () => {
+  test("status: 200, should responds with an article object, containing the properties author, title, article_id, body, topic, created_at, votes, article_img_url and comment_count", () => {
+    return request(app)
+      .get(`/api/articles`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toBeInstanceOf(Array);
+        expect(body).toBeSortedBy("created_at", {
+          descending: true,
```
Comment (hevdevs) [inline __tests__/app.test.js:117]: Great!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -104,3 +105,28 @@ describe("ERROR: GET /api/articles/:article_id", () => {
...
+        });
+        console.log(body);
```
Comment (hevdevs) [inline __tests__/app.test.js:119]: remove console.log before merge

**Review by hevdevs — APPROVED** [review]
All good to merge Omar 🚀🥳

**Pull request merged**